src/contrib/cluster/Akka.Cluster.Sharding/PersistentShard.cs

```diff
@@ -45,29 +93,39 @@ public class PersistentShard : Shard
         /// <param name="extractShardId">TBD</param>
         /// <param name="handOffStopMessage">TBD</param>
         public PersistentShard(
+            IActorContext context,
+            IPersistentActorContext persistentContext,
+            Action<object> unhandled,
```

Comment:
I'll take a look at it. I've already forked @zbynek001 branch in order to continue porting DData shards from there. Maybe I'll propose some changes to the API - I think those changes will have to be taken into account before release. 

In general we try to achieve scala traits trough composition, when inheritance is not enough. For persistence this was somewhat mitigated in the past, but I guess we cannot delay it if we want to move forward.